s3/public.go

```diff
@@ -142,16 +142,30 @@ func (storage *PublishedStorage) PutFile(path string, sourceFilename string) err
 	}
 	defer source.Close()
 
-	err = storage.putFile(path, source)
+	err = storage.putFile(path, source, "")
```

Comment:
Probably I am misunderstanding the issue, but shouldn't we calculate MD5 of the content here and use it when putting the file?

Reply:
As far as I remember, hashes matter only for files from the pool, since the lists are recreated anyway; files in the pool are only being uploaded from `LinkFromPool`, so I left this part without a change.

Of course, it would be a bit more future-proof in case the rest of the code changes and pool files start being uploaded here too, but right now it would be calculating MD5 only for that value to never being used again.